helix-core/src/indent.rs

```diff
@@ -186,106 +188,291 @@ pub fn indent_level_for_line(line: RopeSlice, tab_width: usize) -> usize {
     len / tab_width
 }
 
-/// Find the highest syntax node at position.
-/// This is to identify the column where this node (e.g., an HTML closing tag) ends.
-fn get_highest_syntax_node_at_bytepos(syntax: &Syntax, pos: usize) -> Option<Node> {
-    let tree = syntax.tree();
+struct AddedIndent {
```

Comment:
Can you please add some comments on these structs? I think at least the core stuff should be documented.